### PR TITLE
NickAkhmetov/CAT-775 - EPIC builders boilerplate, migrate API client

### PR DESCRIPTION
**Pull request by NickAkhmetov**
This PR:
- Moves the API client from the `portal-ui` repo to this repo. This will allow us to deduplicate the visualization generation code from the user-templates-api
- Adds more test coverage to the client code
- Sets up the boilerplate for the EPIC builders so we can put together those view configs once there are ingested examples in dev

### CODE REVIEW

File: src/portal_visualization/client.py [hunk 1]
```diff
@@ -0,0 +1,610 @@
...
+        self.entity_api_endpoint = entity_api_endpoint
+
+        self._elasticsearch_endpoint = elasticsearch_endpoint
+        self._portal_index_path = portal_index_path
+
+        self._soft_assay_endpoint = soft_assay_endpoint
+        self._soft_assay_endpoint_path = soft_assay_endpoint_path
+
+        self.elasticsearch_url = f"{elasticsearch_endpoint}{portal_index_path}"
+        self.soft_assay_url = f"{soft_assay_endpoint}/{soft_assay_endpoint_path}"
```
Comment (john-conroy) [inline src/portal_visualization/client.py:94]: Everything here mostly the same as the `client.py` in `portal-ui` outside of passing the configuration values to the constructor?

File: src/portal_visualization/client.py [hunk 2]
```diff
@@ -0,0 +1,610 @@
...
+            else {}
+        )
+        return headers
+
+    def _clean_headers(self, headers):
+        if "Authorization" in headers:
+            headers["Authorization"] = "REDACTED"
+        return headers
+
+    def _request(self, url, body_json=None):
```
Comment (john-conroy) [inline src/portal_visualization/client.py:109]: Is the next action to create a new lib for the duplicated logic between the `portal-ui` and here?

**Review by john-conroy — APPROVED** [review]
Did you introduce a new local code formatter? Should we add it to the repo to avoid future formatter wars?

File: src/portal_visualization/client.py (hunk 1 above)
Comment (NickAkhmetov) [reply src/portal_visualization/client.py:94]: Yes, no changes to functionality, just switched the environment variables to instance variables.

File: src/portal_visualization/client.py (hunk 2 above)
Comment (NickAkhmetov) [reply src/portal_visualization/client.py:109]: I have a PR up to remove the API client from the portal-ui altogether, relying on the instance from here (i.e. it's how I tested that these changes all work/don't break existing visualizations). Once this PR is merged in and a new release is made, I can finalize that PR; at that point I'll also be able to replace the APIClient in the workspace templates with this one and open a corresponding PR there.

**Pull request merged**